src/index.ts

```diff
@@ -0,0 +1,257 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as extend from 'extend';
+import * as util from 'util';
+import {Writable} from 'stream';
+const logging = require('@google-cloud/logging');
+// var Writable = require('stream').Writable;
+
+// Map of Stackdriver logging levels.
+const BUNYAN_TO_STACKDRIVER: {[key: number]: string} = {
+  60: 'CRITICAL',
+  50: 'ERROR',
+  40: 'WARNING',
+  30: 'INFO',
+  20: 'DEBUG',
+  10: 'DEBUG',
+};
+
+/*!
+ * Key to use in the Bunyan payload to allow users to indicate a trace for the
+ * request, and to store as an intermediate value on the log entry before it
+ * gets written to the Stackdriver logging API.
+ */
+export const LOGGING_TRACE_KEY = 'logging.googleapis.com/trace';
+
+/*!
```

Comment:
What's with the exclamation points?

Reply:
This was from the old js code. I haven't changed this before. I have now replaced with standard /** */ style.

Reply:
We just learned that this keeps comments around after running `tsc` :) . Probably not a big deal, but good to know